### DISCUSSION TITLE
Changing Coherence Threshold during ISCE2 Stack Processing

### DISCUSSION THREAD

**Discussion by BrieCorsa** [isce-framework/isce2#775] [Q&A]
Can anyone tell me what the default spatial and temporal coherence thresholds are set to during isce2 stack processing? I have scoured the user manual and searched online without success. Is there a particular flag that I can use to change those default parameters? (For example, I would like to use a coherence threshold of 0.2 and compare the final results with another isce2 stack run that uses coherence threshold of 0.1). 
Thank you for any guidance!

**Comment by EJFielding** [comment]
Hi Brie, the ISCE2 processing does not apply any coherence thresholds. Those are applied in MintPy. You can change them with MintPy configuration parameters.

**Reply by BrieCorsa** [reply]
Hi @EJFielding Thank you so much for your reply- and sorry this is coming back up months later. Doesn't there have to be a coherence threshold for unwrapping interferograms? Do you know what that is defaulted to in ISCE2 stack processing or how to change that threshold, if possible? I will additionally make those threshold adjustments in Mintpy for time series processing- thank you again for getting back to me!

**Reply by BrieCorsa** [reply]
Or perhaps there is a coherence mask/filter that gets applied to the results and that can be adjusted to allow more pixels in?

**Reply by EJFielding** [reply]
The default unwrapping algorithm in ISCE2 is SNAHPU, and SNAPHU unwraps everything without any coherence threshold. All of the coherence thresholds are applied in MintPy. There are two main thresholds in MintPy, the spatial coherence threshold on which interferograms to use (e.g., `mintpy.network.minCoherence     = 0.2`), and the final temporal coherence threshold (e.g., `mintpy.networkInversion.minTempCoh  = auto #[0.0-1.0], auto for 0.7, min temporal coherence for mask`). When you display the results from MintPy with `view.py` it automatically applies the default temporal coherence mask. You can make a different version of the temporal coherence mask with a different threshold with `generate_mask.py -m threshold` utility in MintPy.
